plugins/kubernetes-backend/src/service/KubernetesFetcher.ts

```diff
@@ -105,61 +105,68 @@ export class KubernetesClientBasedFetcher implements KubernetesFetcher {
     this.logger = logger;
   }
 
-  fetchObjectsByServiceId(
+  fetchObjectsForService(
     serviceId: string,
     clusterDetails: ClusterDetails,
     objectTypesToFetch: Set<KubernetesObjectTypes>,
+    labelSelector: string,
```

Comment:
when you reach a number of params like this, it's often clearer for callers to have a single object with named keys to pass in instead of separate unnamed arguments